libs/api/libbpf_program.cpp

```diff
@@ -271,6 +284,18 @@ bpf_program__set_expected_attach_type(struct bpf_program* program, enum bpf_atta
     program->attach_type = *_get_ebpf_attach_type(type);
 }
 
+enum bpf_prog_type
+bpf_program__get_type(const struct bpf_program* program)
+{
+    return _get_bpf_program_type(&program->program_type);
+}
+
+void
+bpf_program__set_type(struct bpf_program* program, enum bpf_prog_type type)
+{
+    program->program_type = *_get_ebpf_program_type(type);
```

Comment:
If input program type is anything other than XDP or BIND, `_get_ebpf_program_type` will return nullptr and this line will dereference nullptr.

Reply:
Fixed